From: Tejas Vijay Kanfade <tkanfade@qti.qualcomm.com>
Date: Thu, 24 Sep 2026 16:16:07 +0530
Subject: configure: add configurable path options for card-defs,
 backend-conf and ACDB

Add --with-card-def-file, --with-backend-conf-file and --with-acdb-path
configure options so that the paths to card-defs.xml, backend_conf.xml
and the ACDB data directory can be set at build time.

Defaults preserve the existing upstream behaviour:
  card-defs.xml      -> /etc/card-defs.xml
  backend_conf.xml   -> /etc/backend_conf.xml
  ACDB data          -> /etc/acdbdata/

Required for QLI/Debian packaging where audioreach-conf moves
card-defs.xml to /etc/audioreach/ and ACDB data to
/usr/share/audioreach/acdbdata/.

Validated on iq-9075-evk with QLI build.

Origin: https://github.com/AudioReach/audioreach-graphmgr/pull/150

Signed-off-by: Tejas Vijay Kanfade <tkanfade@qti.qualcomm.com>
---
 configure.ac                      | 29 +++++++++++++++++++++++++++++
 plugins/tinyalsa/test/Makefile.am |  2 +-
 service/Makefile.am               |  2 +-
 snd_parser/Makefile.am            |  2 +-
 4 files changed, 32 insertions(+), 3 deletions(-)

Index: pkg-audioreach-graphmgr/configure.ac
===================================================================
--- pkg-audioreach-graphmgr.orig/configure.ac
+++ pkg-audioreach-graphmgr/configure.ac
@@ -45,6 +45,35 @@ AC_ARG_WITH([glib],
       AC_HELP_STRING([--with-glib],
          [enable glib, Build against glib. Use this when building for HLOS systems which use glib]))
 
+AC_ARG_WITH([card-def-file],
+    AS_HELP_STRING([--with-card-def-file=FILE],
+        [path to card-defs.xml [default=/etc/card-defs.xml]]),
+    [AGM_CARD_DEF_FILE="$withval"],
+    [AGM_CARD_DEF_FILE="/etc/card-defs.xml"])
+
+AC_ARG_WITH([backend-conf-file],
+    AS_HELP_STRING([--with-backend-conf-file=FILE],
+        [path to backend_conf.xml [default=/etc/backend_conf.xml]]),
+    [AGM_BACKEND_CONF_FILE="$withval"],
+    [AGM_BACKEND_CONF_FILE="/etc/backend_conf.xml"])
+
+AC_ARG_WITH([acdb-path],
+    AS_HELP_STRING([--with-acdb-path=DIR],
+        [path to ACDB data directory [default=/etc/acdbdata/]]),
+    [AGM_ACDB_PATH="$withval"],
+    [AGM_ACDB_PATH="/etc/acdbdata/"])
+
+AC_SUBST([AGM_CARD_DEF_FILE])
+AC_SUBST([AGM_BACKEND_CONF_FILE])
+AC_SUBST([AGM_ACDB_PATH])
+
+AC_ARG_WITH([card-def-native],
+    AS_HELP_STRING([--with-card-def-native=PATH],
+        [path to native card-defs xml (default /etc/card-defs-native.xml)]),
+    [card_def_native=$withval],
+    [card_def_native="/etc/card-defs-native.xml"])
+AC_SUBST([CARD_DEF_FILE_NATIVE], [$card_def_native])
+
 # Optional ALSALIB build support
 AC_ARG_ENABLE([alsalib],
   [AS_HELP_STRING([--enable-alsalib], [Enable ALSA AGM plug-in build (default: no)])],
Index: pkg-audioreach-graphmgr/plugins/tinyalsa/test/Makefile.am
===================================================================
--- pkg-audioreach-graphmgr.orig/plugins/tinyalsa/test/Makefile.am
+++ pkg-audioreach-graphmgr/plugins/tinyalsa/test/Makefile.am
@@ -10,7 +10,7 @@ if USE_G_STR_FUNC
 AM_CFLAGS += -Dstrlcpy=g_strlcpy -Dstrlcat=g_strlcat
 endif
 
-AM_CFLAGS += -DBACKEND_CONF_FILE=\"/etc/backend_conf.xml\"
+AM_CFLAGS += -DBACKEND_CONF_FILE=\"@AGM_BACKEND_CONF_FILE@\"
 lib_LTLIBRARIES      = libagmmixer.la
 libagmmixer_la_SOURCES   = agmmixer.c
 libagmmixer_la_CFLAGS = $(AM_CFLAGS)
Index: pkg-audioreach-graphmgr/service/Makefile.am
===================================================================
--- pkg-audioreach-graphmgr.orig/service/Makefile.am
+++ pkg-audioreach-graphmgr/service/Makefile.am
@@ -28,7 +28,7 @@ libagm_la_SOURCES = $(agm_sources)
 libagm_la_LIBADD = -ltinyalsa @OSAL_LIBS@ @GSL_LIBS@ @ATS_LIBS@ -lsndcardparser
 libagm_la_LDFLAGS = -L$(top_builddir)/snd_parser/.libs
 
-libagm_la_CFLAGS = $(AM_CFLAGS) -DACDB_PATH=\"/etc/acdbdata/\" -DACDB_DELTA_FILE_PATH="/etc/acdbdata/delta"
+libagm_la_CFLAGS = $(AM_CFLAGS) -DACDB_PATH=\"@AGM_ACDB_PATH@\" -DACDB_DELTA_FILE_PATH=\"@AGM_ACDB_PATH@delta\"
 libagm_la_CFLAGS += -DCARD_STATE_UNSUPPORTED
 
 if USE_DEFAULT_ACDB_PATH
Index: pkg-audioreach-graphmgr/snd_parser/Makefile.am
===================================================================
--- pkg-audioreach-graphmgr.orig/snd_parser/Makefile.am
+++ pkg-audioreach-graphmgr/snd_parser/Makefile.am
@@ -11,7 +11,7 @@ lib_includedir = $(includedir)/sndparser
 
 AM_CFLAGS = -I $(srcdir)/inc -I $(top_srcdir)/service/inc/public
 AM_CFLAGS += -Wno-unused-parameter
-AM_CFLAGS += -DCARD_DEF_FILE=\"/etc/card-defs.xml\"
+AM_CFLAGS += -DCARD_DEF_FILE=\"@AGM_CARD_DEF_FILE@\"
 AM_CFLAGS += -DCARD_DEF_FILE_NATIVE=\"/etc/card-defs-native.xml\"
 AM_CFLAGS += -Wl,-z,defs
 
